### PR TITLE
Compare images using raw data

### PR DESCRIPTION
**Pull request by DarkOoze**
Improves the performance of image comparison by using raw data instead of Color structs.

Reading the image to the buffer line by line to prevent the array getting placed on the LOH.
Did not save the measurements I did, but from memory I think this lowered the time from around 80ms to 10ms per image pair.

This adds some dependencies and complexity, not sure what you think of that.
Made a smaller edit with improvements if you rather not add more dependencies I can make a pull-request of that:
https://github.com/DarkOoze/ScreenToGif/tree/local-counter

P.S. My first pull-request, please be gentle 😄.

### CODE REVIEW

**Comment by NickeManarin** [comment]
Hi, sorry for the delay.

Right now, I'm trying to use less dependencies possible. To enable a simple single executable for the portable mode.
This will change when I switch to .NET 5.

File: ScreenToGif/ImageUtil/Gif/LegacyEncoder/PixelUtilOld.cs [hunk 1]
```diff
@@ -16,6 +16,8 @@ public class PixelUtilOld
         private readonly Bitmap _source = null;
         private BitmapData _bitmapData = null;
 
+        readonly object lockObject = new object();
```
Comment (NickeManarin) [inline ScreenToGif/ImageUtil/Gif/LegacyEncoder/PixelUtilOld.cs:19]: private readonly object _lockObject = new object();

Underlines for private variables.

**Comment by DarkOoze** [comment]
Refactored the method to remove the dependencies and the lock object.

**Comment by NickeManarin** [comment]
I'm back on testing this PR and I found an issue.
In a test project, the resulting process got me a different number of frame than the current code (using same settings).

Current:
image

Proposed:
image

image

**Comment by DarkOoze** [comment]
After a couple of tries I was able to reproduce it and found the issue.

The Current implementation don't compare the last pair. Is that intended?

```` csharp
for (var i = 0; i < Project.Frames.Count - 2; i++)
{
````

**Comment by NickeManarin** [comment]
The line below gets Frame + 1, so for the last iteration, there's no next frame to compare.

Looks like the difference is in the image comparison method (I'm still trying to understand).
Oh btw, I just noticed that the image comparison method is using an old PixelUtil utility.
With the newer PixelUtil, it's possible to reduce the delay with the `Unlock()` method by using `UnlockBitsWithoutCommit()` instead (since nothing is being changed, just compared).

**Comment by NickeManarin** [comment]
I noticed that on a test project, frames 2 and 3 have a different equal pixel count.

Current: 222532
Proposed: 222528

Some other frames had the same pixel counts.
I'm still not sure which one is the correct.

[46 frames.zip](https://github.com/NickeManarin/ScreenToGif/files/6340663/46.frames.zip)

**Comment by DarkOoze** [comment]
Are you sure? I'm getting the same result in both versions.

0 1 222554
1 2 222528
2 3 222532
3 4 222528
4 5 222528

**Comment by NickeManarin** [comment]
I was mistaken. Your code is working fine :)

The current code should have been `i < Project.Frames.Count - 1` instead.

**Pull request merged**